apis_core/apis_entities/tables.py

```diff
@@ -71,7 +72,7 @@ def render_name(self, record, value):
 
         class Meta:
             model = caching.get_ontology_class_of_name(entity)
-            fields = default_cols
+            fields = ["detail", "edit"] + default_cols if is_authenticated else ["detail"] + default_cols
```

Comment:
A cleaner way of checking for the right permissions – without inline `if`s (which I personally am not a fan of because they make the code harder to read) and without redundancy – might be to add:

```
if not is_authenticated:
    exclude = ["edit"]
```

to the `Meta` class.

Reply:
This is a good point, I considered that but your comment made me re-think why I did that overall. I agree with that it could (and should!) be more readable, but actually, I think that using fields here is semantically flawed from the beginning and was a mistake from my side anyway. 

Because: 

- fields are fields that come from the model - but the template columns are not part of the model(s)
    - side note: therefore I would protect against possible clashes with model-fields that might be named edit or detail by renaming the columns "_edit" and "_detail" and setting a verbose_name on the columns (would use EditView and DetailView as you suggested @koeaw)
- I used fields here, for the sole purpose to force the order of the columns  (by inserting the column names before  the existing ones). 
- a better way (more semantically clear way) to achieve that ordering would be to use the `sequence = ("detail", "edit", "...")` attribute in the Meta-Class. - see [tables2 docs](https://django-tables2.readthedocs.io/en/latest/pages/swapping-columns.html)

So I would suggest to be very expressive by setting the meta class like this: 

```python 

class Meta:
    model = caching.get_ontology_class_of_name(entity)
    fields =  default_cols 
    if is_authenticated:
        sequence = ('_detail', '_edit', '...') # note: the '...' is special syntax here to add all other columns (fields) in the order they where defined. See the table2 docs mentioned above (same link)
    else: 
       sequence = ('_detail', '...')
```

I think we can't get rid of all redundancy, because: as "_edit" and "_detail" are not fields present in the models, they need to be added to the `fields` attribute or the `sequence` attribute anyway, before I can use `exclude`, as you suggested. So when using `fields` I would need to always add both (for example in default_cols) and then remove "_edit" conditionally, as you suggested. Otherwise, Django throws an error if the template column does not exist or a field with that name.

Reply:
Just a quick note: 

> (would use EditView and DetailView as you suggested @koeaw)

What I actually meant was that the _words shown to users_ should read "Edit" (-> edit this data item) and "View" (-> view the full item referred to by this row). But to clarify internally what the columns point to, i.e. (which) Django views, sounds like a good idea as well.

"Show" was the only other word I could think of for the eye column, but I think "View" fits better semantically. + It should be possible to prevent potential confusion over "View" as verbose name vs. "View" for Django views with the help of code comments for devs.

Reply:
There is no advantage to using `EditView` over `Edit`, only disadvantages from a UI/UX POV.

Interfaces should be well structured, easy to understand, easy to navigate.

Individual elements/parts shouldn't be distracting, take up unnecessary space or interfere with a user's flow in other ways. These two columns are navigational aids that use up space that was previously reserved for the thing that users are actually concerned with (content), so they shouldn't draw even more attention. There is no way around forcing the columns to be wider than their contents of _1 icon_ because we can't have _no_ header, but to broaden them artificially by making the headers even longer (even worse, by using formatting users don't encounter IRL – camel case is not a thing in natural language) when short words are perfectly fine or even self-explanatory does not help users. The suggestion to give the columns a fixed width, and make them _the same width_ (including the word in the header), is part of that, too.

Reply:
I think using the terminology `EditView` or `DetailView` in the user interface is just confusing. If you take github as an example, they simply use `Edit` for the comments we are writing and not `EditView` or `EditForm`. I think the table columns describe *actions* and actions are verbs. So IMO it should be `Edit` and `View`.

But I also think that the table header could be left empty - we are using the same icons in other places without a description what they mean. I would be happy without a table header for those two columns, but there should definitely be a tooltip when hovering over the icons.

Reply:
Alright, I would draft it now without headers, as this should solve both issues (spacing and naming). 

And in a tooltip we can be more precise what we mean ('edit entity' and 'view entity detail') or whatever.